pkg/table/contextimpl/table.go

```diff
@@ -47,10 +47,18 @@ func NewTableContextImpl(sctx sessionctx.Context) *TableContextImpl {
 	}
 }
 
-// TxnRecordTempTable record the temporary table to the current transaction.
-// This method will be called when the temporary table is modified in the transaction.
-func (ctx *TableContextImpl) TxnRecordTempTable(tbl *model.TableInfo) tableutil.TempTable {
-	return ctx.vars().GetTemporaryTable(tbl)
+// AlternativeAllocators implements the AllocatorContext interface
+func (ctx *TableContextImpl) AlternativeAllocators(tbl *model.TableInfo) (allocators autoid.Allocators, ok bool) {
+	// Use an independent allocator for global temporary tables.
+	if tbl.TempTableType == model.TempTableGlobal {
+		if tempTbl := ctx.vars().GetTemporaryTable(tbl); tempTbl != nil {
+			if alloc := tempTbl.GetAutoIDAllocator(); alloc != nil {
+				return autoid.NewAllocators(false, alloc), true
```

Comment:
Why `autoid.NewAllocators(false`?  Shouldn't it be `autoid.NewAllocators(tbl.SepAutoInc()`

What happen for this case, it uses a different autoid key encoding?

```
create global temporary table t (id int) on commit delete rows AUTO_ID_CACHE=1  
```

Reply:
B.t.w, why we did not set Allocators for temp table in the past?
This function call increase object allocation by 1 op/s

Reply:
The code is from https://github.com/pingcap/tidb/pull/39041/files#diff-ccb2771ef1931990c0d0ce3703def7dd53bd21d00a8f6b40b59ca0115f776a60R1550 
This PR does not change any code logic.

However, I think it's fine for the current code because the id allocation for memory table is pure in memory and does not share with other sessions.